hal/sdio_api.h

```diff
@@ -0,0 +1,163 @@
+/** \addtogroup hal */
```

Comment:
What is the reason for having a new API instead of extending already existing BlockDevice? Maybe in a similar manner as SDBlockDevice.

Reply:
Or are there any plans to get this driver to Mbed OS one day?

Reply:
One day there will be plans to add this to MbedOS. The original code comes from 2 different implementations. @orenc17 did some work to align them. I understand the reason of the existence of this repository as a place to work out the APIs and the generic part. Having the different implementations in the same tree will allow easier changes and testing.